src/olympia/files/utils.py

```diff
@@ -134,7 +135,25 @@ def __init__(self, path, data=''):
             with open(self.path) as fobj:
                 data = fobj.read()
 
-        self.data = json.loads(unicodehelper.decode(data))
+        lexer = JsLexer()
+
+        json_string = ''
+
+        # Run through the JSON and remove all comments, then try to read
+        # the manifest file.
+        # Note that Firefox and the Web Extension spec only allow for
+        # line comments (starting with `//`), not block comments (starting with
+        # `/*`). We strip out both in AMO because the linter will flag the
+        # block-level comments explicitly as an error (so the developer can
+        # change them to line-level comments).
+        #
+        # But block level comments are not allowed. We just flag them elsewhere
+        # (in the linter).
+        for name, token in lexer.lex(data):
+            if name not in ['blockcomment', 'linecomment']:
```

Comment:
Just for the sake of having at least one nitpick, it's best to use tuples for not changing data structures like this, e.g

```python
if name not in ('blockcomment', 'linecomment'):
    # ...
```

saves a few bits in memory and maybe a few more cpu cycles (because a list is stored in a way that allws for appending/extending it, tuples are immutable) but that's really just micro-optimization here.